src/components/Execution/SingleNodeAggregation.jsx

```diff
@@ -49,43 +55,72 @@ const SingleNodeAggregation = ({ node }) => {
     setIsWorking(false)
   }, [node, client, nbShares, pretrained, setIsWorking])
 
-  return (
-    <div className="selected-single-node">
-      <div className="single-node-title">
-        {node && (node.label ? node.label : node.id)}
+  return isLoading ? (
+    <Spinner size="xxlarge" middle />
+  ) : (
+    <div className="card">
+      <div className="card-title">
+        <b>Single node aggregation</b>
+      </div>
+      <div>
+        Performs the distributed aggregation protocol using different processes
+        of the same instance: the instance compute a local model, sends shares
+        to itself, aggregates them and then does the final aggregation.
       </div>
       <div>
-        <Label htmlFor="single-node-shares">Number of shares: </Label>
-        <Input
-          value={nbShares}
-          onChange={e => setNbShares(e.target.value)}
-          id="single-node-shares"
+        <Label htmlFor="single-node-selector">
+          Select the node performing the execution:{' '}
+        </Label>
+        <SelectBox
+          id="single-node-selector"
+          options={options}
+          name="Select a node"
+          onChange={e => setSingleNode(e.value)}
         />
       </div>
-      <FormControlLabel
-        label="Use pretrained model?"
-        control={
-          <Switch
-            checked={pretrained}
-            onChange={() => setPretrained(old => !old)}
-            name="Use pretrained model?"
+      <div className="spacer-sm" />
+      <div className="selected-single-node">
+        <div className="single-node-title">
+          {node && (node.label ? node.label : node.id)}
+        </div>
+        <div>
+          <Label htmlFor="single-node-shares">Number of shares: </Label>
+          <Input
+            value={nbShares}
+            onChange={e => setNbShares(e.target.value)}
+            id="single-node-shares"
           />
-        }
-      />
-      <Button
-        className="todo-remove-button"
-        //theme="danger"
-        iconOnly
-        label="Launch execution"
-        busy={isWorking}
-        disabled={isWorking}
-        onClick={handleLaunchExecution}
-        extension="narrow"
-      >
-        Launch execution
-      </Button>
+        </div>
+        <FormControlLabel
+          label="Use pretrained model?"
+          control={
+            <Switch
+              checked={pretrained}
+              onChange={() => setPretrained(old => !old)}
+              name="Use pretrained model?"
+            />
+          }
+        />
+        <Button
+          className="todo-remove-button"
+          //theme="danger"
+          iconOnly
+          label="Launch execution"
+          busy={isWorking}
+          disabled={!node || isWorking}
+          onClick={handleLaunchExecution}
+          extension="narrow"
+        >
+          Launch execution
+        </Button>
+      </div>
     </div>
   )
 }
 
-export default SingleNodeAggregation
+export default queryConnect({
```

Comment:
I initially missed that, but it shouldn't be necessary: why did you add it?

Reply:
It's gone in the recent commits